[YAML] test_get_traction_info_empty - Exception: Query failed

### Background / motivation

Both tests in test_get_traction_info.py fail. It seems to be a sporadic failure. We should get to the bottom of the failures, and stabilize this test to be more reliable.

```text
=========================== short test summary info ============================
FAILED pdr_backend/analytics/test/test_get_traction_info.py::test_get_traction_info_main_mainnet - Exception: Query failed. Url: https://v4.subgraph.sapphire-mainnet.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph. Return code is 503

            {
                predictSubscriptions(skip: 0, first: 1000 , where: {predictContract_: {id_in: ["0x123"]}, timestamp_gt: 1698883200, timestamp_lt: 1699142400}) {
                    id
                    txId
                    timestamp
                    user {
                        id
                    }
                    predictContract {
                        id
                        token {
                            id
                            name
                            lastPriceValue
                            nft{
                                nftData {
                                key
                                value
                                }
                            }
                        }
                    }
                }
            }
FAILED pdr_backend/analytics/test/test_get_traction_info.py::test_get_traction_info_empty - Exception: Query failed. Url: https://v4.subgraph.sapphire-mainnet.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph. Return code is 503

        {
            tokens(where: {
                nft_: {
                    owner: "0x4ac2e51f9b1b0ca9e000dfe6032b24639b172703"
                }
            }) {
                id
            }
        }
======= 2 failed, 330 passed, 3 skipped, 9 warnings in 108.16s (0:01:48) =======
```

See [this log](https://github.com/oceanprotocol/pdr-backend/actions/runs/7479549022/job/20356983138?pr=510) for full traceback. 

Note: this failure is sporadic. So far, I've (Trent) only seen it once, in the log above (Jan 10, 2024). 

- Q: Was this issue was likely introduced in the tweaks for network override as part of PR https://github.com/oceanprotocol/pdr-backend/pull/510 ?
- A: No, because both tests that fail have the network override _off_. (I double-checked this with more testing)

--- a/pdr_backend/analytics/test/test_get_traction_info.py
+++ b/pdr_backend/analytics/test/test_get_traction_info.py
@@ -1,6 +1,7 @@
 from unittest.mock import Mock, patch
 
 import polars as pl
+import pytest
 from enforce_typing import enforce_types
 
 from pdr_backend.analytics.get_traction_info import get_traction_info_main
@@ -88,3 +89,12 @@ def test_get_traction_info_empty(tmpdir, capfd):
     assert (
         "No records found. Please adjust start and end times." in capfd.readouterr().out
     )
+
+    with patch("requests.post") as mock_post:
+        mock_post.return_value.status_code = 503
+        # don't actually sleep in tests
+        with patch("time.sleep"):
+            with pytest.raises(Exception):
+                get_traction_info_main(ppss, st_timestr, fin_timestr, "parquet_data/")
+
+    assert mock_post.call_count == 3

--- a/pdr_backend/subgraph/core_subgraph.py
+++ b/pdr_backend/subgraph/core_subgraph.py
@@ -1,3 +1,4 @@
+import time
 from typing import Dict
 
 import requests
@@ -18,13 +19,17 @@ def query_subgraph(
     @return
       result -- e.g. {"data" : {"predictContracts": ..}}
     """
-    request = requests.post(subgraph_url, "", json={"query": query}, timeout=timeout)
-    if request.status_code != 200:
+    response = requests.post(subgraph_url, "", json={"query": query}, timeout=timeout)
+    if response.status_code != 200:
         # pylint: disable=broad-exception-raised
         if tries < SUBGRAPH_MAX_TRIES:
+            time.sleep(((tries + 1) / 2) ** (2) * 10)
             return query_subgraph(subgraph_url, query, tries + 1)
+
         raise Exception(
-            f"Query failed. Url: {subgraph_url}. Return code is {request.status_code}\n{query}"
+            f"Query failed. Url: {subgraph_url}. Return code is {response.status_code}\n{query}"
         )
-    result = request.json()
+
+    result = response.json()
+
     return result

